### PR TITLE
Add new Ops to Azure Databricks for KFP: secretscope, workspaceitem & dbfsblock

### PR DESCRIPTION
**Pull request by magencio**
Adds new Ops to Azure Databricks for Kubeflow Pipelines:
* CreateSecretScopeOp, to create a secret scope in Databricks.
* DeleteSecretScopeOp, to delete a secret scope created with CreateSecretScopeOp.
* ImportWorkspaceItemOp, to import an item into a Databricks Workspace.
* DeleteWorkspaceItemOp, to delete an item imported with ImportWorkspaceItemOp.
* CreateDbfsBlockOp, to create Dbfs block in Databricks.
* DeleteDbfsBlockOp, to delete Dbfs block created with CreateDbfsBlockOp.

Adds new pipeline samples to show the usage of these new Ops.

Fixes some typos in existing Ops.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2817)

### CODE REVIEW

**Comment by magencio** [comment]
@googlebot I fixed it.

**Comment by chaitanyak52** [comment]
@googlebot I fixed it.

**Comment by chaitanyak52** [comment]
@googlebot I fixed it.

**Comment by magencio** [comment]
@googlebot I fixed it.

**Comment by chaitanyak52** [comment]
@googlebot I fixed it.

**Comment by chaitanyak52** [comment]
@googlebot I consent

**Comment by creddy123** [comment]
@googlebot I consent

**Comment by numerology** [comment]
Thanks! @magencio 

Quick question: do you mind becoming a Kubeflow member [here](https://github.com/kubeflow/internal-acls/blob/master/github-orgs/kubeflow/org.yaml) and add yourself (probably your colleagues as well) as the OWNER under `samples/contrib/azure-samples`? Thanks

File: samples/contrib/azure-samples/databricks-pipelines/databricks_notebook_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+def import_workspace_item(item_name):
+    current_path = Path(__file__).parent
+    notebook_file_name = current_path.joinpath("notebooks", "ScalaExampleNotebook")
+    notebook = open(notebook_file_name).read().encode("utf-8")
+    notebook_base64 = base64.b64encode(notebook)
+    return databricks.ImportWorkspaceItemOp(
+        name="importworkspaceitem",
+        item_name=item_name,
+        content=notebook_base64,
+        path="/Users/alejacma@microsoft.com/ScalaExampleNotebook",
```
Comment (numerology) [inline samples/contrib/azure-samples/databricks-pipelines/databricks_notebook_pipeline.py]: Do you want to wipe out your personal email here and add some comment to remind users to replace it?

File: samples/contrib/azure-samples/databricks-pipelines/databricks_notebook_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        num_workers=2
+    )
+
+def create_job(job_name, cluster_id):
+    return databricks.CreateJobOp(
+        name="createjob",
+        job_name=job_name,
+        existing_cluster_id=cluster_id,
+        notebook_task={
+            "notebook_path": "/Users/alejacma@microsoft.com/ScalaExampleNotebook"
```
Comment (numerology) [inline samples/contrib/azure-samples/databricks-pipelines/databricks_notebook_pipeline.py]: ditto.

File: samples/contrib/azure-samples/kfp-azure-databricks/setup.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 setup(
     name='kfp-azure-databricks',
-    version='0.1.0',
+    version='0.2.0',
```
Comment (numerology) [inline samples/contrib/azure-samples/kfp-azure-databricks/setup.py]: Maybe grab version number from __init__ to keep a single source of truth?

File: samples/contrib/azure-samples/kfp-azure-databricks/tests/test_dbfsblock_op.py [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+            res = CreateDbfsBlockOp(
+                name="createdbfsitem",
+                block_name=block_name,
+                path=path,
+                data=data
+            )
+            
+            self.assert_res(res, expected_spec)
+
+        kfp.compiler.Compiler()._compile(my_pipeline)
```
Comment (numerology) [inline samples/contrib/azure-samples/kfp-azure-databricks/tests/test_dbfsblock_op.py]: `_compile()` is being deprecated, maybe switch to `_create_workflow()`?

File: samples/contrib/azure-samples/databricks-pipelines/databricks_notebook_pipeline.py (hunk 1 above)
Comment (magencio) [reply samples/contrib/azure-samples/databricks-pipelines/databricks_notebook_pipeline.py]: Done

File: samples/contrib/azure-samples/databricks-pipelines/databricks_notebook_pipeline.py (hunk 2 above)
Comment (magencio) [reply samples/contrib/azure-samples/databricks-pipelines/databricks_notebook_pipeline.py]: Done. Now the notebook gets the user as a parameter

File: samples/contrib/azure-samples/kfp-azure-databricks/setup.py (hunk 3 above)
Comment (magencio) [reply samples/contrib/azure-samples/kfp-azure-databricks/setup.py]: Done

File: samples/contrib/azure-samples/kfp-azure-databricks/tests/test_dbfsblock_op.py (hunk 4 above)
Comment (magencio) [reply samples/contrib/azure-samples/kfp-azure-databricks/tests/test_dbfsblock_op.py]: Done

**Comment by numerology** [comment]
/lgtm
/approve
Thanks @magencio

File: .gitignore [hunk 5]
```diff
@@ -33,6 +33,7 @@ frontend/test/ui/visual-regression/screenshots/screen
 .idea/
 .ijwb/
 *.iml
+.history/*
```
Comment (numerology) [inline .gitignore]: If you can clean up your local repo without changing .gitignore then you do not need root OWNER approval :)

Comment (magencio) [reply .gitignore]: You are right! I didn't realize that was part of a commit. I removed that line from .gitignore. So no more need for root OWNER  approval. Thx!

**Pull request merged**